server/controllers/jobs_controller.go

```diff
@@ -29,102 +37,22 @@ type JobsController struct {
 	StatsScope               tally.Scope
 }
 
-type ProjectInfoKeyGenerator struct{}
-
-func (g ProjectInfoKeyGenerator) Generate(r *http.Request) (string, error) {
-	projectInfo, err := newProjectInfo(r)
-
-	if err != nil {
-		return "", errors.Wrap(err, "creating project info")
-	}
-
-	return projectInfo.String(), nil
-}
-
-type pullInfo struct {
-	org  string
-	repo string
-	pull int
-}
-
-func (p *pullInfo) String() string {
-	return fmt.Sprintf("%s/%s/%d", p.org, p.repo, p.pull)
-}
-
-type projectInfo struct {
-	projectName string
-	workspace   string
-	pullInfo
-}
-
-func (p *projectInfo) String() string {
-	return fmt.Sprintf("%s/%s/%d/%s/%s", p.org, p.repo, p.pull, p.projectName, p.workspace)
-}
-
-func newPullInfo(r *http.Request) (*pullInfo, error) {
-	org, ok := mux.Vars(r)["org"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no org in route")
-	}
-	repo, ok := mux.Vars(r)["repo"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no repo in route")
-	}
-	pull, ok := mux.Vars(r)["pull"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no pull in route")
-	}
-	pullNum, err := strconv.Atoi(pull)
-	if err != nil {
-		return nil, err
-	}
-
-	return &pullInfo{
-		org:  org,
-		repo: repo,
-		pull: pullNum,
-	}, nil
-}
-
-// Gets the PR information from the HTTP request params
-func newProjectInfo(r *http.Request) (*projectInfo, error) {
-	pullInfo, err := newPullInfo(r)
-	if err != nil {
-		return nil, err
-	}
-
-	project, ok := mux.Vars(r)["project"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no project in route")
-	}
-
-	workspace, ok := mux.Vars(r)["workspace"]
-	if !ok {
-		return nil, fmt.Errorf("Internal error: no workspace in route")
-	}
-
-	return &projectInfo{
-		pullInfo:    *pullInfo,
-		projectName: project,
-		workspace:   workspace,
-	}, nil
-}
-
 func (j *JobsController) getProjectJobs(w http.ResponseWriter, r *http.Request) error {
-	projectInfo, err := newProjectInfo(r)
-	if err != nil {
-		j.respond(w, logging.Error, http.StatusInternalServerError, err.Error())
+	jobID, ok := mux.Vars(r)["job-id"]
```

Comment:
Should probably re-use the logic in `JobIDKeyGenerator`. It is essentially the same thing

Reply:
Done!